conda_forge_tick/migrators.xsh

```diff
@@ -140,43 +168,27 @@ class Version(Migrator):
 
     def migrate(self, recipe_dir, attrs, hash_type='sha256'):
         # Render with new version but nothing else
+        $VERSION = attrs['new_version']
+        with indir(recipe_dir):
+            with open('meta.yaml', 'r') as f:
+                text = f.read()
+        url = re.search('  url:.*?\n(    -.*\n?)*', text).group()
+        if 'cran.r-project.org/src/contrib' in url:
+            $VERSION = $VERSION.replace('_', '-')
         with indir(recipe_dir):
             for f, p, n in self.patterns:
                 p = eval_version(p)
                 n = eval_version(n)
                 replace_in_file(p, n, f)
             with open('meta.yaml', 'r') as f:
                 text = f.read()
-            # If we can't parse the meta_yaml then jump out
-            meta_yaml = parsed_meta_yaml(text)
-            # If the parser returns None, then we didn't read the meta.yaml
-            # TODO: How we didn't fail at 01 on this recipe is mysterious
-            if meta_yaml is None:
-                attrs['bad'] = '{}: failed to read meta.yaml\n'.format($PROJECT)
-                return False
-            source_url = meta_yaml.get('source', {}).get('url')
-            if not source_url:
-                attrs['bad'] = '{}: missing url\n'.format($PROJECT)
-                return False
-            if isinstance(source_url, list):
-                for url in source_url:
-                    if 'Archive' not in url:
-                        source_url = url
-                        break
-            if 'cran.r-project.org/src/contrib' in source_url:
-                $VERSION = $VERSION.replace('_', '-')
-
-        # now, update the feedstock to the new version
-        source_url = eval_version(source_url)
-        try:
-            hash = hash_url(source_url, hash_type)
-        except urllib.error.HTTPError:
-            attrs['bad'] = '{}: hash failed at {}\n'.format(
-                meta_yaml.get('package', {}).get('name', 'UNKOWN'), source_url)
-            return False
-
-        new_patterns = tuple(self.patterns) + tuple(self.more_patterns)
-        with indir(recipe_dir), ${...}.swap(HASH_TYPE=hash_type, HASH=hash, SOURCE_URL=source_url):
```

Comment:
The reason to use `${...}.swap()` is so that the environment is returned to the same values that it originally had when the function is called. This is how I would prefer it.  the current function seems leaky. You can add $VERSION or any other envvars you'd likt to this too!

Reply:
HASH_TYPE, HASH, and SOURCE_URL aren't used anywhere now, as the hash type and hash are written directly in the patterns returned by `get_hash_patterns`, so I don't think the swap is needed. I suppose this also means `eval_version` isn't necessary either.

Reply:
Right, but you are setting version, so I think swapping in/out `$VERSION` is sensible.  Unless like you said, you think you can get rid of that code somehow